e2e/playwright/benchmark/hot-path.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { expect, test } from '@e2e/playwright/zoo-test'
+import { DefaultLayoutPaneID } from '@src/lib/layout/configs/default'
+
+/**
+ * Benchmark tests for the first things an user would do in the app.
+ * This is meant to represent user behavior, and changes to that are owned by our group of
+ * mechanical engineers. See .github/CODEOWNERS for more details.
+ */
+
+test.describe('Hot path', () => {
+  test(`Draw a circle and extrude it`, async ({
+    page,
+    homePage,
+    scene,
+    editor,
+    toolbar,
+    cmdBar,
+  }) => {
+    // Open an empty project
+    await page.setBodyDimensions({ width: 1500, height: 1000 })
+    await homePage.goToModelingScene()
+    await scene.settled(cmdBar)
+
+    // Default layout
+    await toolbar.openPane(DefaultLayoutPaneID.FeatureTree)
+    await toolbar.openPane(DefaultLayoutPaneID.Code)
+
+    // Mouse helpers
+    const [clickCenter] = scene.makeMouseHelpers(0.5, 0.5, { format: 'ratio' })
+    const [clickABitOffCenter] = scene.makeMouseHelpers(0.55, 0.45, {
+      format: 'ratio',
+    })
+
+    // Default step timeout
+    const timeout = 500
+
+    await test.step('Start sketch', async () => {
+      await toolbar.startSketchBtn.click()
+      await page.waitForTimeout(timeout)
+      await clickABitOffCenter()
+      await page.waitForTimeout(timeout)
+    })
+
+    await test.step('Draw circle', async () => {
+      await toolbar.circleBtn.click()
+      await page.waitForTimeout(timeout)
+      await clickCenter()
+      await page.waitForTimeout(timeout)
+      await clickABitOffCenter()
+      await page.waitForTimeout(timeout)
+    })
+
+    await test.step('Exit sketch, expect feature tree update', async () => {
+      await toolbar.exitSketchBtn.click()
+      await scene.settled(cmdBar)
+      await editor.expectEditor.toContain('circle(')
+      await expect(
+        await toolbar.getFeatureTreeOperation('sketch001', 0)
+      ).toBeVisible()
+    })
+
+    await test.step('Click Extrude and select circle profile', async () => {
+      await toolbar.extrudeButton.click()
+      await cmdBar.expectState({
+        stage: 'arguments',
+        currentArgKey: 'sketches',
+        currentArgValue: '',
+        headerArguments: {
+          Profiles: '',
+        },
+        highlightedHeaderArg: 'Profiles',
+        commandName: 'Extrude',
+      })
+      await clickCenter()
+    })
+
+    await test.step('Click Continue, see error about missing args on review page', async () => {
+      await cmdBar.progressCmdBar()
+      await cmdBar.expectState({
+        stage: 'review',
+        headerArguments: {
+          Profiles: '1 profile',
+        },
+        commandName: 'Extrude',
+        reviewValidationError:
+          'semantic: Either `length` or `to` parameter must be provided for extrusion.',
+      })
+      await page.waitForTimeout(timeout)
+    })
```

Comment:
Seeing this error right away without any wrong input is annoying and I'd like this to change soon.

But this is exactly what a test like this would capture. I don't want to go crazy on this approach until we validate that the mechanical engineer group can actually find meaning in user behavior represented in such a file, and if we can keep track of that over time. Worth giving it a shot?